### PR TITLE
fix(typings): use correct ignoreDuplicates option, add create test

### PR DESCRIPTION
**Pull request by JacobLey**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you update the typescript typings accordingly (if applicable)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)?

### Description of change

`onDuplicate` is not a real option. Should be `ignoreDuplicates` https://github.com/sequelize/sequelize/blob/main/lib/dialects/abstract/query-generator.js#L196

Added tests and extended `returning` option

### CODE REVIEW

File: types/lib/model.d.ts [hunk 1]
```diff
@@ -749,7 +754,7 @@ export interface BulkCreateOptions<TAttributes = any> extends Logging, Transacti
   individualHooks?: boolean;
 
   /**
-   * Ignore duplicate values for primary keys? (not supported by postgres)
+   * Ignore duplicate values for primary keys?
```
Comment (JacobLey) [inline types/lib/model.d.ts:757]: While the `ignoreDuplicates` constant is undefined for postgres, the option maps directly to `onConflictDoNothing` https://github.com/sequelize/sequelize/blob/main/lib/dialects/abstract/query-generator.js#L197 which is defined for postgres https://github.com/sequelize/sequelize/blob/main/lib/dialects/postgres/index.js#L48

File: types/lib/model.d.ts [hunk 2]
```diff
@@ -1975,7 +1980,7 @@ export abstract class Model<TModelAttributes extends {} = any, TCreationAttribut
   public static create<M extends Model>(
     this: ModelStatic<M>,
     values: M['_creationAttributes'],
-    options: CreateOptions<M['_attributes']> & { returning: false }
+    options: Omit<CreateOptions<M['_attributes']>, 'returning'> & { returning: false }
```
Comment (JacobLey) [inline types/lib/model.d.ts:1983]: Should we give `ignoreDuplicates` similar treatment?

A conflict with "do nothing" could potentially end up in the instance not being saved to the database. Might be better to invalidate the response (`void`) and force a reload of the instance if necessary

**Comment by JacobLey** [comment]
Pretty sure test failure is unrelated...

File: types/lib/model.d.ts (hunk 2 above)
Comment (JacobLey) [reply types/lib/model.d.ts:1983]: Actually I like that, let's do it. Better to be correct on the safe side.

**Review by papb — APPROVED** [review]
Great work! Thank you very much!

**Pull request merged**